adapter/internal/discovery/xds/rate_limiter_cache_test.go

```diff
@@ -782,3 +783,70 @@ func TestAddSubscriptionLevelRateLimitPolicy(t *testing.T) {
 
 	assert.Equal(t, expectedPolicies, rlsPolicyCache.metadataBasedPolicies)
 }
+
+func TestSubscriptionRateLimitPolicyEventDataHandling(t *testing.T) {
+	policy := types.SubscriptionPolicy{
+		Name: "policyToCheckDeletion",
+		DefaultLimit: &types.SubscriptionDefaultLimit{
+			QuotaType: "requestCount",
+			RequestCount: &types.SubscriptionRequestCount{
+				RequestCount: 6100,
+				TimeUnit:     "hour",
+			},
+		},
+		Organization: "org2",
+		RateLimitCount: 61,
+		RateLimitTimeUnit: "min",
+		StopOnQuotaReach: false,
+	}
+	AddSubscriptionLevelRateLimitPolicy(policy)
+	_, ok := rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization]
+	if !ok {
+		t.Errorf("Expected policies for organization %q to exist in the cache, but it doesn't", policy.Organization)
+		return
+	}
+	updatedPolicy := types.SubscriptionPolicy{
+		Name: "policyToCheckDeletion",
+		DefaultLimit: &types.SubscriptionDefaultLimit{
+			QuotaType: "requestCount",
+			RequestCount: &types.SubscriptionRequestCount{
+				RequestCount: 6000,
```

Comment:
shall we have a comment on the field where we have updated? It will be easy to read then.